zubhub_backend/zubhub/activities/models.py

```diff
@@ -0,0 +1,161 @@
+import uuid
+from django.db import models
+from django.contrib.auth import get_user_model
+from django.utils.text import slugify
+from django.utils import timezone
+from math import floor
+
+Creator = get_user_model()
+
+class Category(MP_Node):
+    name = models.CharField(max_length=100, unique=True)
+    description = models.CharField(max_length=1000, blank=True, null=True)
+    slug = models.SlugField(unique=True, max_length=1000)
+    search_vector = SearchVectorField(null=True)
+
+    node_order_by = ['name']
+
+    class Meta:
+        verbose_name_plural = "categories"
+        indexes = (GinIndex(fields=["search_vector"]), )
+
+    def __str__(self):
+        return self.name
+
+    def save(self, *args, **kwargs):
+        if self.slug:
+            pass
+        else:
+            uid = str(uuid.uuid4())
+            uid = uid[0:floor(len(uid) / 6)]
+            self.slug = slugify(self.name) + "-" + uid
+        super().save(*args, **kwargs)
+
+class Image(models.Model):
+    image_url = models.URLField(max_length=1000)
+    public_id = models.CharField(max_length=1000, null=True, blank=True)
```

Comment:
https://docs.djangoproject.com/en/2.0/ref/models/fields/#null
Recommends that we Avoid using [null](https://docs.djangoproject.com/en/2.0/ref/models/fields/#django.db.models.Field.null) on string-based fields such as [CharField](https://docs.djangoproject.com/en/2.0/ref/models/fields/#django.db.models.CharField) and [TextField](https://docs.djangoproject.com/en/2.0/ref/models/fields/#django.db.models.TextField). If a string-based field has null=True, that means it has two possible values for “no data”: NULL, and the empty string.

We can use blank=True if we want to allow blank value submission via forms.